test/common/testlib.py

```diff
@@ -1437,8 +1437,9 @@ def new_machine(
                 self.network = network
             networking = self.network.host(restrict=restrict, forward=forward or {})
             machine = machine_class(verbose=opts.trace, networking=networking, image=image, **kwargs)
-            if opts.fetch and not os.path.exists(machine.image_file):
-                machine.pull(machine.image_file)
+            image_file = machine.image_file  # type: ignore[attr-defined]
```

Comment:
This is indeed not defined, sending a bots PR. This seems easy enough to fix.

Reply:
Right, but it still doesn't help much due due to the dynamic class. I did try to declare `image_file` locally in bots/, but it didn't help.

Reply:
I.e. we can't declare `machine` as being a `VirtMachine` class, as it could be changed to a different class (which anaconda does). Well, on second thought we perhaps *can* declare it such and ignore the wrong type.

If you have any other idea how to deal with this, I'm happy to take out that commit and just keep the static-code fix.